Cubical/ZCohomology/CohomologyRings/KleinBottle.agda

```diff
@@ -0,0 +1,572 @@
+     base-neutral-eq (zero        ∷ one         ∷ []) = cong (base 2) (pres1 ϕ₂∘ψ₂str) ∙ base-neutral _
+     base-neutral-eq (zero        ∷ suc (suc m) ∷ []) = refl
+     base-neutral-eq (one         ∷ zero        ∷ []) = cong (base 1) (pres1 ϕ₁str) ∙ base-neutral _
+     base-neutral-eq (one         ∷ suc m       ∷ []) = refl
+     base-neutral-eq (suc (suc n) ∷ m           ∷ []) = refl
+
+     base-add-eq : _
+     base-add-eq (zero        ∷ zero        ∷ []) a b = base-add _ _ _ ∙ cong (base 0) (sym (pres· ϕ₀str _ _))
+     base-add-eq (zero        ∷ one         ∷ []) a b = base-add _ _ _ ∙ cong (base 2) (sym (pres· ϕ₂∘ψ₂str _ _))
+     base-add-eq (zero        ∷ suc (suc m) ∷ []) a b = +H*IdR _
+     base-add-eq (one         ∷ zero        ∷ []) a b = base-add _ _ _ ∙ cong (base 1) (sym (pres· ϕ₁str _ _))
+     base-add-eq (one         ∷ suc m       ∷ []) a b = +H*IdR _
+     base-add-eq (suc (suc n) ∷ m           ∷ []) a b = +H*IdR _
+
+  -----------------------------------------------------------------------------
+  -- Morphism on ℤ[x]
+
+    ℤ[x,y]→H*-𝕂²-pres1 : ℤ[x,y]→H*-𝕂² (1Pℤ) ≡ 1H*
+    ℤ[x,y]→H*-𝕂²-pres1 = refl
+
+    ℤ[x,y]→H*-𝕂²-pres+ : (x y : ℤ[x,y]) → ℤ[x,y]→H*-𝕂² (x +Pℤ y) ≡ ℤ[x,y]→H*-𝕂² x +H* ℤ[x,y]→H*-𝕂² y
+    ℤ[x,y]→H*-𝕂²-pres+ x y = refl
+
+    --           Explanation of the product proof
+    --
+    --           -------------------------------------------------------
+    --           | (0,0) | (0,1) | (0,m+2) | (1,0) | (1,m+1) | (n+2,m) |
+    -- -----------------------------------------------------------------
+    -- | (0,0)   |   H⁰  |   H⁰  |    0    |   H⁰  |    0    |    0    |
+    -- -----------------------------------------------------------------
+    -- | (0,1)   |  Sym  |   0₄  |    0    |   0₃  |    0    |    0    |
+    -- -----------------------------------------------------------------
+    -- | (0,m+2) | ==========================================> triv    |
+    -- -----------------------------------------------------------------
+    -- | (1,0)   |  Sym  |  Sym  |    0    |   ★  |    0    |    0    |
+    -- -----------------------------------------------------------------
+    -- | (1,m+1) | ==========================================> triv    |
+    -- -----------------------------------------------------------------
+    -- | (n+2,m) | ==========================================> triv    |
+    -- -----------------------------------------------------------------
+
+    -- ★ : prove that ϕ₁(1) ⌣ ϕ₁(1) ≡ 0
+
+    open pres⌣
+
+
+    ϕ₀-gen : (n : ℕ) → (f : coHom n KleinBottle) → ϕ₀ (pos 1) ⌣ f ≡ f
+    ϕ₀-gen n = ST.elim (λ _ → isProp→isSet (GroupStr.is-set (snd (coHomGr n KleinBottle)) _ _))
+                       (λ f → cong ∣_∣₂ (funExt (λ x → rUnitₖ n (f x))))
+
+    -- note that the proof might be simpliale by adding a second partition on T
+    -- side, though it might complicated a bunch of things
```

Comment:
I'm confused by this comment... Does extra partitioning simplify things or not?

Reply:
I don't know, that's why there is a comment